api/practice/controllers/practice.js

```diff
@@ -5,50 +5,70 @@
  * to customize this controller
  */
 
-const { sanitizeEntity } = require('strapi-utils');
-const util = require('util');
-
-function getUrlVariable(url, query) {
-  url = url.replace(/.*?\?/, '');
-  url = url.replace(/_&_/, '_%26_');
-  var vars = url.split('&');
-  for (var i = 0; i < vars.length; i++) {
-    var pair = vars[i].split('=');
-    if (decodeURIComponent(pair[0]) == query) {
-      return decodeURIComponent(pair[1]);
+module.exports = {
+  // findByKeyword returns all practices with text matching "keyword" in either the title, subtitle,
+  // or tags associated with the practice
```

Comment:
Nice job on making this a more general search and likely more efficient in the process.  I recommend changing this comment to reflect the expanded scope.